phac-nml/irida-next
# Fix: Allow for unlimited sample filter using filter by name or id dialog [ENHC0010016]

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixes issue caused by using a `get` request to filter the table by list of puids or names.  Get requests are limited to 2,048 characters which prevented this.  Changed the form to use a `post` request.  To facilitate this, I needed to:

1. Update the search form to surround the entire fields and table.  This was required because the pagination requests needed to use the data in the search fields and therefore must also be submitted as a `post`
2. Updated the select all to make a copy of the search form and update the URL for search, and then request the submission of this new form.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

No UI changes were required.

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Go to any project that has samples
2. Try "Filter by sample name or id", I created a text file with 300 random sample names and a couple from the project I was in to ensure that the filter was working and could handle a large amount of text.
3. Try pagination with the filter
4. Ensure selection and select all is working

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
